blockindex/contractstaking/cache.go

```diff
@@ -78,6 +78,9 @@ func (s *contractStakingCache) CandidateVotes(ctx context.Context, candidate add
 		}
 		bt := s.mustGetBucketType(bi.TypeIndex)
 		if featureCtx.FixContractStakingWeightedVotes {
+			if s.config.CalculateVoteWeight == nil {
+				return nil, errors.New("calculate vote weight function is not set")
+			}
```

Comment:
if we add this check, it should be added at creation time, when the func is passed to `Newxxx`

Reply:
ok